### PR TITLE
LSP: Various Surround With refactoring fixes.

### PR DESCRIPTION
**Pull request by dbalek**
- Problem with too complex surround with menu fixed - original menu narrowed down couple of most common items only, the rest is available on generic `Surround with ...` invocation.
- Surround with try produces broken code - fixed
- Entries for `try/catch` and `try/catch/finally` added

### CODE REVIEW

File: java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/SurroundWithHint.java [hunk 1]
```diff
@@ -127,14 +132,21 @@
...
-                    codeActions.add(codeAction);
+                    if (TO_SHOW.contains(codeTemplate.getAbbreviation())) {
+                        codeActions.add(codeAction);
+                    }
+                    items.add(new QuickPickItem(label, null, text, false, codeAction));
                 }
             }
         }
+        if (items.size() > codeActions.size()) {
+            codeActions.add(createCodeAction(Bundle.DN_SurroundWith(DOTS), CodeActionKind.RefactorRewrite, COMMAND_SURROUND_WITH, items));
```
Comment (sdedic) [inline java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/SurroundWithHint.java:148]: I'd rather create a separate message, or make DOTS localizable.

**Review by JaroslavTulach — APPROVED** [review]

**Review by sdedic — APPROVED** [review]

File: ide/editor.codetemplates/src/org/netbeans/lib/editor/codetemplates/CodeTemplateCompletionProvider.java [hunk 2]
```diff
@@ -89,8 +89,9 @@ private static boolean isAbbrevDisabled(JTextComponent component) {
     public static class Collector implements CompletionCollector {
 
         private static final String SELECTED_TEXT_VAR = "${0:$TM_SELECTED_TEXT}";
-        private static final Pattern SNIPPET_VAR_PATTERN = Pattern.compile("\\$\\{\\s*([-\\w]++)([^}]*)?}");
-        private static final Pattern SNIPPET_HINT_PATTERN = Pattern.compile("([-\\w]++)(?:\\s*=\\s*(\\\"([^\\\"]*)\\\"|\\S*))?");
+        private static final Pattern SNIPPET_VAR_PATTERN = Pattern.compile("\\$\\{\\s*([-\\w]++)((?:\\s*[-\\w]++(?:\\s*=\\s*(?:\\\"[^\\\"]*\\\"|[-\\w]++))?)*\\s*)?}");
```
[the remark this answers is not in the document]
Comment (JaroslavTulach) [reply ide/editor.codetemplates/src/org/netbeans/lib/editor/codetemplates/CodeTemplateCompletionProvider.java:92]: I didn't bother. But having few tests showing what the pattern matches and what it rejects wouldn't hurt future readers.

File: java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/SurroundWithHint.java (hunk 1 above)
Comment (dbalek) [reply java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/SurroundWithHint.java:148]: Fixed.

File: ide/editor.codetemplates/src/org/netbeans/lib/editor/codetemplates/CodeTemplateCompletionProvider.java [hunk 3]
```diff
@@ -100,8 +101,9 @@ public boolean collectCompletions(Document doc, int offset, Completion.Context c
                     description = CodeTemplateApiPackageAccessor.get().getSingleLineText(ct);
                 }
                 String label = html2text(description.trim());
+                String sortText = String.format("%04d%s", 1650, "fore".equals(ct.getAbbreviation()) ? label.substring(0, 3) : label);
```
[the remark this answers is not in the document]
Comment (dbalek) [reply ide/editor.codetemplates/src/org/netbeans/lib/editor/codetemplates/CodeTemplateCompletionProvider.java:104]: Ugly hack to put foreach above all other variants of for templates offered in code completion list.

File: ide/editor.codetemplates/src/org/netbeans/lib/editor/codetemplates/CodeTemplateCompletionProvider.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (dbalek) [reply ide/editor.codetemplates/src/org/netbeans/lib/editor/codetemplates/CodeTemplateCompletionProvider.java:92]: Quoted-string syntax disallows `\"` IMHO. However, the pattern is used to parse NB built-in templates (using only "simple" quoted-strings).

**Pull request merged**